src/mavlink-router/mainloop.cpp

```diff
@@ -758,44 +758,58 @@ void Mainloop::_handle_pipe()
     if (num_read > 0) {
         cmd[num_read] = 0;
         log_debug("Pipe read %ld bytes: %s", num_read, cmd);
-        // Op UDP Name IP Port Eavesdropping
-        std::vector<std::string> a;
-        std::string command = cmd;
-        char *pch = strtok(cmd, " ");
-        while (pch != nullptr) {
-            a.emplace_back(pch);
-            pch = strtok(nullptr, " \n");
-        }
-
-        if (a.size() == 2 && a[0] == "remove") {
-            _remove_dynamic_endpoint(a[1]);
-            return;
-        }
 
-        if (a.size() != 6 || a[0] != "add" || a[1] != "udp") {
-            return;
-        }
-        std::string name = a[2];
-        std::string address = a[3];
-        int port = atoi(a[4].c_str());
-        if (port <= 0) {
-            return;
-        }
-
-        auto pipecmd = _pipe_commands.find(name);
-        if (pipecmd != _pipe_commands.end() && pipecmd->second == command) {
-            auto ep = _dynamic_endpoints.find(name);
-            if (ep != _dynamic_endpoints.end()) {
-                ep->second->reset_expire_timer();
-                return;
-            }
+        // if more than one command separated by an end of
+        // line was written in the pipe, separate each command
+        std::vector<char*> command_vector;
```

Comment:
Can you combine the 2 loops? You store the commands in a vector and then iterate over them. This can be done in a single loop w/o the need of a vector.
You can't use `strtok` then, but `strchr` instead.